vertx-mongo-client/src/test/java/io/vertx/ext/mongo/DistinctTest.java

```diff
@@ -0,0 +1,188 @@
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class DistinctTest extends MongoTestBase {
+
+  protected MongoClient mongoClient;
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    JsonObject config = getConfig();
+    mongoClient = MongoClient.createNonShared(vertx, config);
+    CountDownLatch latch = new CountDownLatch(1);
+    dropCollections(latch);
+    awaitLatch(latch);
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    mongoClient.close();
+    super.tearDown();
+  }
+
+  @Test
+  public void testDistinctInteger() {
+    String collection = randomCollection();
+    insertDocs(collection, 10, onSuccess(inserted -> {
+      mongoClient.distinct(collection, "num", Integer.class.getName(), onSuccess(distincted -> {
+        assertEquals(1, distincted.size());
+        assertEquals(new Integer(123), distincted.getInteger(0));
+        testComplete();
+      }));
+    }));
+    await();
+  }
+
+  @Test
+  public void testDistinctString() {
+    String collection = randomCollection();
+    insertDocs(collection, 10, onSuccess(inserted -> {
+      mongoClient.distinct(collection, "foo", String.class.getName(), onSuccess(distincted -> {
+        assertEquals(10, distincted.size());
+        testComplete();
+      }));
+    }));
+    await();
+  }
+
+  @Test
+  public void testDistinctBadResultClass() {
+    String collection = randomCollection();
+    insertDocs(collection, 10, onSuccess(inserted -> {
+      mongoClient.distinct(collection, "foo", Object.class.getName(), onFailure(failure -> {
+        testComplete();
+      }));
+    }));
+    await();
+  }
+
+  @Test
+  public void testDistinctStringBadFormat() {
+    String collection = randomCollection();
+    insertDocs(collection, 10, onSuccess(inserted -> {
+      mongoClient.distinct(collection, "foo", Integer.class.getName(), onFailure(failure -> {
+        testComplete();
+      }));
+    }));
+    await();
+  }
+
+  @Test
+  public void testDistinctUnexistentString() {
+    String collection = randomCollection();
+    insertDocs(collection, 10, onSuccess(inserted -> {
+      mongoClient.distinct(collection, "unexist", String.class.getName(), onSuccess(distincted -> {
+        assertEquals(0, distincted.size());
+        testComplete();
+      }));
+    }));
+    await();
+  }
+
+  @Test
+  public void testDistinctBatchString() throws Exception {
+    String collection = randomCollection();
+    int numDocs = 10;
+    CountDownLatch latch = new CountDownLatch(numDocs);
+    insertDocs(collection, numDocs, onSuccess(inserted -> {
+      mongoClient.distinctBatch(collection, "foo", String.class.getName(), onSuccess(distincted -> {
+        assertNotNull(distincted);
+        latch.countDown();
+      }));
+    }));
+    awaitLatch(latch);
+  }
+
+  protected void dropCollections(CountDownLatch latch) {
```

Comment:
These are actually duplicate methods from MongoClientTestBase. It seems that those methods could be pulled up to MongoTestBase to avoid the duplication. It will mean passing mongoClient into dropCollections but I would prefer it to the copied methods